sttp/client/src/main/scala/endpoints/sttp/client/Endpoints.scala

```diff
@@ -0,0 +1,112 @@
+package endpoints.sttp.client
+
+import java.net.URI
+
+import endpoints.algebra
+import endpoints.Tupler
+import com.softwaremill.sttp
+
+import scala.language.higherKinds
+
+/**
+  * An interpreter for [[endpoints.algebra.Endpoints]] that builds a client issuing requests using
+  * a sttp’s [[com.softwaremill.sttp.SttpBackend]].
+  *
+  * Doest not support streaming responses for now
+  *
+  * @param host     Base of the URL of the service that implements the endpoints (e.g. "http://foo.com")
+  * @param backend  The underlying backend to use
+  * @tparam R       The monad wrapping the response. It is defined by the backend
+  */
+class Endpoints[R[_]](host: String, val backend: sttp.SttpBackend[R, Nothing]) extends algebra.Endpoints with Urls with Methods {
+
+  /**
+    * A function that, given an `A` and a request model, returns an updated request
+    * containing additional headers
+    */
+  type RequestHeaders[A] = (A, sttp.Request[_, Nothing]) => sttp.Request[_, Nothing]
```

Comment:
can we type alias `sttp.Request[_, Nothing]`? It seems to clutter the code

Reply:
I’m not familiar with sttp, what’s the role of these two type parameters?

Reply:
The first type is the type of the body in the request: `String` `Array[Int]` `Inputstream`...

The second allows to specify if the response should be streamed and should match the second parameter of the backend. For strict backend it uses `Nothing`, and for streaming backend like akka-http it will be `Source[ByteString, Any]`

Since it's the first implementation and I'm not really familiar with the streaming part, I prefered focusing on the strict backend.